### PR TITLE
First table migration for testing purposes

### PR DESCRIPTION
**Pull request by arivepr**
Adding this just for testing purposes.

Proving the service will actually create the tables when running for the first time, otherwise pulling them in. 

Screen Shot 2022-07-27 at 10 44 24 AM

### CODE REVIEW

File: rest/models/base.go [hunk 1]
```diff
@@ -2,11 +2,13 @@ package models
...
   "time"
+  "gorm.io/gorm"
 )
 
 // Generic Struct used throughout models in this service.
 type BaseModel struct {
   Id uint
   CreatedAt time.Time
   UpdatedAt time.Time
+  DeletedAt gorm.DeletedAt
```
Comment (Hyperkid123) [inline rest/models/base.go:13]: Is it possible to use `time.Time` as well? The issue with the gorm type is that indexes the field as well and causes issue with OpenAPI within spec.

Comment (arivepr) [reply rest/models/base.go:13]: For sure! if I understand this correctly, you want me to go from gorm.DeletedAt, to just a simple time stamp as well?

File: rest/database/db.go [hunk 2]
```diff
@@ -29,6 +28,17 @@ func Init() {
...
 	DB, err = gorm.Open(dialector, &gorm.Config{})
 
+	// Migration/Creation of data tables for DB
+	if !DB.Migrator().HasTable(&models.UserIdentity{}) {
+		DB.Migrator().CreateTable(&models.UserIdentity{})
+	}
+	if !DB.Migrator().HasTable(&models.FavoritePage{}) {
+		DB.Migrator().CreateTable(&models.FavoritePage{})
+	}
+	if !DB.Migrator().HasTable(&models.LastVisitedPage{}) {
```
Comment (Hyperkid123) [inline rest/database/db.go:38]: This model does not exist yet.

File: rest/database/db.go [hunk 3]
```diff
@@ -29,6 +28,17 @@ func Init() {
 
 	DB, err = gorm.Open(dialector, &gorm.Config{})
 
+	// Migration/Creation of data tables for DB
+	if !DB.Migrator().HasTable(&models.UserIdentity{}) {
+		DB.Migrator().CreateTable(&models.UserIdentity{})
```
Comment (Hyperkid123) [inline rest/database/db.go:33]: This model does not exist yet

File: rest/database/db.go (hunk 2 above)
Comment (arivepr) [reply rest/database/db.go:38]: Yeah this file change went past me when I stashed the other changes, I'm changing it back rn

**Comment by arivepr** [comment]
Please excuse the Merge, had to wrestle git for a bit 😅 😅 @Hyperkid123

File: rest/models/base.go (hunk 1 above)
Comment (Hyperkid123) [reply rest/models/base.go]: Yeah. The gorm.DeletedAt is a bit problematic. We will see how it turns out.

**Pull request merged**